terraform/node/k8s/deploy/main.tf

```diff
@@ -0,0 +1,172 @@
+# ------------------------------------------------------------------------
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+resource "null_resource" "deploy" {
+  connection {
+    type        = "ssh"
+    user        = var.user
+    private_key = var.ssh_key
+    host        = var.host
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      <<-EOF
+      # Make the Terraform fail if any step throws an error
+      set -e
+
+      # Ensure environment is clean
+      echo "LOG: Rerunning cleanup commands in case of cleanup failure in previous run"
+      helm uninstall --debug --namespace amazon-cloudwatch amazon-cloudwatch-operator --ignore-not-found
+      kubectl delete namespace sample-app-namespace --ignore-not-found=true
+      [ ! -e amazon-cloudwatch-agent-operator ] || sudo rm -r amazon-cloudwatch-agent-operator
+      [ ! -e frontend-service-depl.yaml ] || rm frontend-service-depl.yaml
+      [ ! -e remote-service-depl.yaml ] || rm remote-service-depl.yaml
+
+      # Clone and install operator onto cluster
+      echo "LOG: Cloning helm-charts repo"
+      # TODO: Update to latest release version of the code
+      git clone https://github.com/aws-observability/helm-charts -q
+      cd helm-charts/charts/amazon-cloudwatch-observability/
+      git reset --hard b407a9545a03e377703ef965a2ee655aa6df7406
```

Comment:
Are we going to update the git hash `b407a9545a03e377703ef965a2ee655aa6df7406` with new release of helm chart? If that's the case, is it better to define a variable instead?

Reply:
Once we have that, we will use the latest releast instead of the git hash